docs/chapters/04_features.md

```diff
@@ -446,6 +463,134 @@ mutation {
 
+### Reading events
+
+You can also fetch events directly if you need. To do so, there are two kind of queries that have the following structure:
+```graphql
+query {
+  eventsByEntity(entity: <name of entity>, entityID: "<id of the entity>") {
+    selection_field_list
+  }
+}
+
+query {
+  eventsByType(type: <name of event>) {
+    selection_field_list
+  }
+}
+```
+Where:
+- _<name of your entity>_ is the name of the class corresponding to the entity whose events you want to retrieve.
+- _<id of the entity>_ is the ID of the specific entity instance whose events you are interested in. **This is optional**
+- _<name of event>_ is the name of the class corresponding to the event type whose instances you want to retrieve.
+- _selection_field_list_ is a list with the names of the specific fields you want to get as response. See the response example below to know more.
+
+
+#### Examples
+
+```
+  URL: "<graphqlURL>"
+```
+**A) Read all events associated with CartEntity by a specific ID**
+```graphql
+query {
+  eventsByEntity(entity: CartEntity, entityID: "B5") {
+    type entity entityID requestID createdAt value
+  }
+}
+```
+
+**B) Read all events from CartEntity**
+```graphql
+query {
+  eventsByEntity(entity: CartEntity) {
+    type entity entityID requestID createdAt value
+  }
+}
+```
+
+For cases A and B, the response look like this:
+```json
+{
+    "data": {
+        "eventsByEntity": [
+            {
+                "type": "CartItemChanged",
+                "entity": "Cart",
+                "entityID": "B5",
+                "requestID": "7a9cc6a7-7c7f-4ef0-aef1-b226ae4d94fa",
+                "createdAt": "2021-05-12T08:41:13.792Z",
+                "value": {
+                    "productId": "73f7818c-f83e-4482-be49-339c004b6fdf",
+                    "cartId": "B5",
+                    "quantity": 2
+                }
+            }
+        ]
+    }
+}
+```
+
+**C) Query specific events, no matter the entity/es it has assigned.**
+```graphql
+query {
+  eventsByType(type: CartChangedEvent) {
+    type entity entityID requestID createdAt value
+  }
+}
+```
+In this case, for the **eventsByType** endpoint, the response would be:
+```json
+{
+    "data": {
+        "eventsByType": [
+            {
+                "type": "CartItemChanged",
+                "entity": "Cart",
+                "entityID": "5d81a60c-2e27-4fef-9f57-e5ae36519144",
+                "requestID": "7a9cc6a7-7c7f-4ef0-aef1-b226ae4d94fa",
+                "createdAt": "2021-05-12T08:41:13.792Z",
+                "value": {
+                    "productId": "73f7818c-f83e-4482-be49-339c004b6fdf",
+                    "cartId": "5d81a60c-2e27-4fef-9f57-e5ae36519144",
+                    "quantity": 2
+                }
+            }, 
+          ...
+        ]}
+    }
+}
+```
+
+#### Time filters
+Optionally, you can get events in a range of dates for any endpoint of this API. The time filter format is in ISO format, with any precision, for example:
```

Comment:
Remember that in GraphQL, we talk about mutations or queries, not endpoints, as we have only one endpoint for everything.
```suggestion
Optionally, for any of the previous queries, you can include a `from` and/or `to` time filters to get only those events that happened inside that time range. You must use a string with a time in ISO format with any precision you like, for example:
```